### PR TITLE
Adding Support for Azure Environments [untested]

### PR DESCRIPTION
**Pull request by FriedrichWeinmann**
**This PR fixes**

https://github.com/Azure/AzOps/issues/141

Adds support for Azure Environments, by reading a $env:AZURE_ENVIRONMENT environment variable.

> Unable to test this so far, advice on testing the container would be appreciated :)

### CODE REVIEW

File: src/private/Write-AzOpsLog.ps1 [hunk 1]
```diff
@@ -62,7 +62,4 @@ function Write-AzOpsLog {
             }
         }
     }
-
-    end { }
```
Comment (ljtill) [inline src/private/Write-AzOpsLog.ps1:66]: Could we this block in, we'd like to have a consistent code structure across all functions?

File: entrypoint.ps1 [hunk 2]
```diff
@@ -19,17 +19,26 @@ function Logging {
...
             # Create credential
-            $credentials = ($env:AZURE_CREDENTIALS | ConvertFrom-Json)
-            $credential = New-Object System.Management.Automation.PSCredential -ArgumentList $credentials.clientId, ($credentials.clientSecret | ConvertTo-SecureString -AsPlainText -Force)
-
+            $credentials = $env:AZURE_CREDENTIALS | ConvertFrom-Json
+			$credential = [System.Management.Automation.PSCredential]::new($credentials.clientId, ($credentials.clientSecret | ConvertTo-SecureString -AsPlainText -Force))
+			$supportedEnvironments = 'AzureUSGovernment','AzureCloud','AzureChinaCloud'
+			if (-not $env:AZURE_ENVIRONMENT) { $environment = 'AzureCloud' }
+			elseif ($env:AZURE_ENVIRONMENT -notin $supportedEnvironments) {
+				Write-AzOpsLog -Level Error -Topic "env-var" -Message "Illegal azure environment: $env:AZURE_ENVIRONMENT . Supported environments: $($supportedEnvironments -join ",")"
```
Comment (ljtill) [inline entrypoint.ps1:34]: Could we rephrase this to 'Unsupported azure environment'?

File: entrypoint.ps1 [hunk 3]
```diff
@@ -106,10 +115,6 @@ function Initialization {
             exit 1
         }
     }
-
-    end {
```
Comment (ljtill) [inline entrypoint.ps1:110]: Could we this block in, we'd like to have a consistent code structure across all functions?

**Review by ljtill — CHANGES_REQUESTED** [review]

File: entrypoint.ps1 (hunk 3 above)
Comment (FriedrichWeinmann) [reply entrypoint.ps1:110]: Can do that, sure.
Comes at a slight performance cost, so I was instead planning in purging it from all commands on the refactor (can do that in bulk then though, so undoing that)

**Comment by uday31in** [comment]
@FriedrichWeinmann  - can you please update docs https://github.com/Azure/Enterprise-Scale/blob/main/docs/Deploy/setup-github.md
